src/shared/bpf-program.c

```diff
@@ -76,6 +76,10 @@ int bpf_program_load_kernel(BPFProgram *p, char *log_buf, size_t log_size) {
                 return 0;
         }
 
+        /* FIXME: Clang doesn't 0-pad with structured initialization, causing
+         * the kernel to reject the bpf_attr as invalid. Remove memsets once
+         * the compilers we support are fixed. */
+        memset(&attr, 0, sizeof(attr));
```

Comment:
```c
zero(attr);
```

memset() is just too hard to use, people mix up the args all the time

Reply:
thanks, fixed